ocrd_models/ocrd_page_user_methods.py

```diff
@@ -83,6 +83,34 @@ def show(self):
 #
 # Replace the following method specifications with your own.
 
+#
+# export children sorted by index of the childelement
+#
+sort_children_by_index = MethodSpec(name='exportChildren',
+    source=r'''
+    def exportChildren(self, outfile, level, namespaceprefix_='', namespacedef_='xmlns:pc="http://schema.primaresearch.org/PAGE/gts/pagecontent/2019-07-15"', name_='OrderedGroupType', fromsubclass_=False, pretty_print=True):
+        if pretty_print:
+            eol_ = '\n'
+        else:
+            eol_ = ''
+        if self.UserDefined is not None:
+            namespaceprefix_ = self.UserDefined_nsprefix_ + ':' if (UseCapturedNS_ and self.UserDefined_nsprefix_) else ''
+            self.UserDefined.export(outfile, level, namespaceprefix_, namespacedef_='', name_='UserDefined', pretty_print=pretty_print)
+        for Labels_ in self.Labels:
+            namespaceprefix_ = self.Labels_nsprefix_ + ':' if (UseCapturedNS_ and self.Labels_nsprefix_) else ''
+            Labels_.export(outfile, level, namespaceprefix_, namespacedef_='', name_='Labels', pretty_print=pretty_print)
+        for RegionRefIndexed_ in sorted(self.RegionRefIndexed, key=lambda rri: rri.index):
+            namespaceprefix_ = self.RegionRefIndexed_nsprefix_ + ':' if (UseCapturedNS_ and self.RegionRefIndexed_nsprefix_) else ''
+            RegionRefIndexed_.export(outfile, level, namespaceprefix_, namespacedef_='', name_='RegionRefIndexed', pretty_print=pretty_print)
+        for OrderedGroupIndexed_ in sorted(self.OrderedGroupIndexed, key=lambda ogi: ogi.index):
+            namespaceprefix_ = self.OrderedGroupIndexed_nsprefix_ + ':' if (UseCapturedNS_ and self.OrderedGroupIndexed_nsprefix_) else ''
+            OrderedGroupIndexed_.export(outfile, level, namespaceprefix_, namespacedef_='', name_='OrderedGroupIndexed', pretty_print=pretty_print)
+        for UnorderedGroupIndexed_ in sorted(self.UnorderedGroupIndexed, key=lambda ugi: ugi.index):
+            namespaceprefix_ = self.UnorderedGroupIndexed_nsprefix_ + ':' if (UseCapturedNS_ and self.UnorderedGroupIndexed_nsprefix_) else ''
+            UnorderedGroupIndexed_.export(outfile, level, namespaceprefix_, namespacedef_='', name_='UnorderedGroupIndexed', pretty_print=pretty_print)
```

Comment:
```suggestion
       namespaceprefix_ = ''
        if UseCapturedNS_:
            if self.RegionRefIndexed_nsprefix_:
                namespaceprefix_ = self.RegionRefIndexed_nsprefix_ + ':'
            elif self.OrderedGroupIndexed_nsprefix_:
                namespaceprefix_ = self.OrderedGroupIndexed_nsprefix_ + ':'
            elif self.UnorderedGroupIndexed_nsprefix_:
                namespaceprefix_ = self.UnorderedGroupIndexed_nsprefix_ + ':'
        for entry in sorted(self.RegionRefIndexed + self.OrderedGroupIndexed + self.UnorderedGroupIndexed, key=lambda rri: rri.index):
entry.export(outfile, level, namespaceprefix_, namespacedef_='', name_=entry.__class__.__name__[:-4], pretty_print=pretty_print)
```

Reply:
Another worthwhile thing this PR could cover is the problem [as Christian explained](https://github.com/PRImA-Research-Lab/prima-page-viewer/issues/15#issuecomment-623492135), empty groups are not permitted by the schema and must therefore be reduced to mere RegionRefs.

This is relevant because many processors will try to create groups for pages or recursive regions (tables, text regions with constituent text regions), which then may happen to be empty, because either no content was found or some error happened. It would be daunting to always have to recover from that by converting all `OrderedGroup(Indexed)` or `UnorderedGroup(Indexed)` to `RegionRef(Indexed)` with equal `@regionRef` (and `@index`).